http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/internal/S3JavaHttpClientIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+    private void createS3Client() {
+        this.s3Client = S3Client.create();
+    }
+
+    private void createAsyncClientWithJavaHttpClientHTTP1() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP1_1).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    private void createAsyncClientWithJavaHttpClientHTTP2() {
+        SdkAsyncHttpClient javaHttpCient = JavaHttpClientNioAsyncHttpClient.builder().protocol(Protocol.HTTP2).build();
+        createS3AsyncClient(javaHttpCient);
+    }
+
+    /* To run the tests in order, here alphabets are added at the beginning of the names of tests. */
+    @Test
+    public void As3createBucketWithJavaHttpClientHTTP1Test() {
+        createAsyncClientWithJavaHttpClientHTTP1();
+        createBucketTest(s3Async);
+    }
+
+    @Test
+    public void Bs3createBucketWithJavaHttpClientHTTP2Test() {
+        createAsyncClientWithJavaHttpClientHTTP2();
+        createBucketTest(s3Async);
+    }
+
+    @Test
+    public void Cs3PutSmallObjectWithJavaHttpClientHTTP1Test() {
+        createAsyncClientWithJavaHttpClientHTTP1();
+        putObjectTest(s3Async, KEY, VALUE);
+    }
+
+    @Test
+    public void Ds3PutSmallObjectWithJavaHttpClientHTTP2Test() {
+        createAsyncClientWithJavaHttpClientHTTP2();
+        putObjectTest(s3Async, KEY, VALUE);
+    }
+
+    @Test
+    public void Es3PutLargeObjectWithJavaHttpClientHTTP1Test() {
+        createAsyncClientWithJavaHttpClientHTTP1();
+        putObjectTest(s3Async, LARGEKEY, LARGESTRING);
+    }
+
+    @Test
+    public void Fs3PutLargeObjectWithJavaHttpClientHTTP2Test() {
+        createAsyncClientWithJavaHttpClientHTTP2();
+        putObjectTest(s3Async, LARGEKEY, LARGESTRING);
+    }
+
+    @Test
+    public void Gs3GetObjectWithJavaHttpClientHTTP1Test() {
+        createAsyncClientWithJavaHttpClientHTTP1();
+        getObjectTest(s3Async);
+    }
+
+    @Test
+    public void Hs3GetObjectWithJavaHttpClientHTTP2Test() {
+        createAsyncClientWithJavaHttpClientHTTP2();
+        getObjectTest(s3Async);
+    }
+
+    @Test
+    public void Is3GetObjectWithSyncHttpClient() {
+        createS3Client();
+        ResponseBytes responseBytes = s3Client.getObject(GetObjectRequest.builder().bucket(BUCKET).key(KEY).build(),
+                ResponseTransformer.toBytes());
+        assertEquals(responseBytes.asUtf8String(), VALUE);
+    }
+
+    @Test
+    public void Js3DeleteObjectWithJavaHttpClientHTTP1Test() {
+        createAsyncClientWithJavaHttpClientHTTP1();
+        deleteObjectTest(s3Async);
+    }
+
+
+    private static void createBucketTest(S3AsyncClient s3Async) {
+        try (s3Async) {
+            s3Async.createBucket(CreateBucketRequest.builder().bucket(BUCKET).build()).join();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void putObjectTest(S3AsyncClient s3Async, String key, String value) {
+        try (s3Async) {
```

Comment:
I think we discussed this, but we need to use a second client to download the object we just uploaded and ensure that it's all there.

Reply:
Sorry, that was a wrong file, I changed that in another file named S3JavaHttpClientIntegratedTest.java, which should be the real "S3JavaHttpClientIntegrationTest.java". I don't understand why this file just keeps appearing even after I have deleted it. It has been fixed now.